TESTING.md

```diff
@@ -0,0 +1,274 @@
+out. This is most confusing to those who approach Cypress commands as Promises.
+[Commands are not Promises](https://docs.cypress.io/guides/core-concepts/introduction-to-cypress.html#Commands-Are-Not-Promises).
+
+For example, imagine the following Cypress code:
+
+```ts
+cy.get('body')
+  .contains('button', 'Delete Item')
+  .click();
+
+cy.get('[data-testid="MyResult"]').should('contain', 'Success!');
+```
+
+The Cypress Command queue will show `['get', 'contains', 'click', 'get', 'should']`, but it will not
+run these commands immediately. There is also no need to use `.then` to wait for any previous
+condition to be met first. The Cypress run time will do the following:
+
+- Run `$('body')` every 50ms until `$('body')` returns a non-empty jQuery collection
+- Run a function that gets all `button` elements inside of the `body` element that have text content
+  that contains 'Delete Item' and that jQuery collection is not empty
+- Ensure the previous element is clickable (visible, not covered, has actual width/height), waiting
+  50ms to detect animations (waiting for animations to complete), etc. Basically make sure a user
+  could actually click on it.
+- Run `$('[data-testid="MyResult"]')` every 50ms until it returns a non-empty jQuery collection.
+  Usually clicking has side-effects like network requests to the server before the side-effect is
+  reflected in the DOM. Since Cypress implicitly retries, this command will _eventually_ return a
+  non-empty jQuery collection and the author doesn't need to explicitly do anything. Note this
+  selector _could_ return a non-empty jQuery collection instantly, but it might not contain
+  "Success!" yet...
+- Run internal `chai` + `chai-jquery` matchers (`expect($el).to.contain('Success!')`) until an error
+  is no longer thrown. `.should` will rerun the previous command _and_ the assertion until the
+  condition is met or a timeout occurs. This is the secret sauce to the stability of Cypress tests
+  compared to WebDriver. `.should` can also receive a function that will be run until no error is
+  thrown. If an error is never thrown, it will only be run once.
+
+Cypress is very easy to get started with, but there isn't enough documentation about how to scale a
+complicated suite of Cypress tests.
+
+### Component helpers
+
+Component helpers should be written to help with automation. These component helpers are useful for
+a more readable implementation as well as more maintainable test code. Component helpers can also be
+exported by this repository for use in downstream testing (Cypress tests in applications that use
+Canvas Kit). Helpers contain implementation details about how components are composed internally.
+The goal is that if the Component implementation changes, the component helpers should be the only
+code that needs to change to keep tests passing.
+
+Component helpers come in 3 flavors:
+
+- **Starter**: This helper types starts a Cypress chain. The return type is always
```

Comment:
This one always gets me, but "types starts" sounds weird but "types start" sounds about right?

```suggestion
- **Starter**: This helper types start a Cypress chain. The return type is always
```

Reply:
Should it be "This helper type starts..."?